libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
+            _flags.startup_beep = true;
+            play_pattern(SINGLE_BUZZ);
+            return;
+        }
+        // play beeps when system initialization complete
+        if (!_flags.initialize_started) {
+            if (AP_Notify::flags.initialising) {
+                _flags.initialize_started = true;
+            }
+        } else if (!AP_Notify::flags.initialising) {
+            _flags.initialize_done = true;
+            play_pattern(INITIALIZE_BUZZ);
+            return;
+        }
+    }
+
+    // check if radio failsafe status has changed
+    if (_flags.failsafe_radio != AP_Notify::flags.failsafe_radio) {
+        _flags.failsafe_radio = AP_Notify::flags.failsafe_radio;
+        // play beeps to indicate radio status (if initialization completed)
+        if (_flags.initialize_done) {
+            play_pattern(_flags.failsafe_radio ? RADIOLOST_BUZZ : RADIOBACK_BUZZ);
+            return;
+        }
+    }
+
     // check for arming failed event
     if (AP_Notify::events.arming_failed) {
         // arming failed buzz
         play_pattern(SINGLE_BUZZ);
         return;
     }
 
-    if (AP_HAL::millis() - _pattern_start_time < _pattern_start_interval_time_ms) {
-        // do not interrupt playing patterns / enforce minumum separation
+    // do not interrupt playing patterns
+    if (is_playing_pattern()) {
+        return;
+    }
+
+    // if battery failsafe constantly buzz
+    if (AP_Notify::flags.failsafe_battery) {
+        play_pattern(CONSTANT_BUZZ);
```

Comment:
This is a behavioural change - there are people out there expecting a single buzz for battery failsafe.  If you change that those people are going to turn into angry, torch-and-pitchfork-wielding villagers.  Or so I imagine.

i think you'll need a different startup tone.  If you can find something approximating the ToneAlarm startup that would be great.  Rather unlikely, I acknowledge.

Reply:
Via some testing I've found that in Copter v3.6.10 / Plane 3.9.10 (with active buzzer) the voltage failsafe sound is repeated single beeps at about 3 per second, so I'll look to replicate that.

I think an init tone with 3-3-5 beeps may work; I'll see.